### PR TITLE
[12.0][MIG] resource

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Migration scripts for `resource` module.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---Fields in module 'resource'---
+resource     / resource.calendar        / hours_per_day (float)         : NEW
```
Comment (pedrobaeza) [inline addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt]: You should compute this field getting the average of the hours per day. You can call `_onchange_hours_per_day` on post-migration for getting this.

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---Fields in module 'resource'---
+resource     / resource.calendar        / hours_per_day (float)         : NEW 
+resource     / resource.calendar.attendance / day_period (selection)        : NEW required: required, selection_keys: ['afternoon', 'morning'], req_default: function
```
Comment (pedrobaeza) [inline addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:5]: I would filled this depending on the ~~`date_to`~~ `hour_from` field: if before 12:00 a.m, `morning`. `afternoon` elsewhere.

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+resource     / resource.calendar.leaves / time_type (selection)         : NEW selection_keys: ['leave', 'other']
```
Comment (pedrobaeza) [inline addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:8]: I don't have this clear if the default value `leave` is the proper one for getting the same result as in v11. Have you investigated this?

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+# NOTHING TO DO
+
+resource     / resource.calendar.leaves / tz (selection)                : DEL selection_keys: function
+resource     / resource.calendar        / tz (selection)                : NEW required: required, selection_keys: function, req_default: function
+# Done: post-migration: filled tz from leave_ids.tz
+
+resource     / resource.resource        / tz (selection)                : NEW required: required, selection_keys: function, req_default: function
+# Done: post-migration: filled tz from user_id.tz
```
Comment (pedrobaeza) [inline addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:16]: There are other possible resources to take into account when doing the migration scripts. For example, workcenters for `mrp` module. But this is to be done in that module :wink:

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 2 above)
Comment (MiquelRForgeFlow) [reply addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:5]: and how I put 12:00 a.m in postgres?

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 3 above)
Comment (MiquelRForgeFlow) [reply addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:8]: It should be handled in hr_holidays module? The resource.calendar.leaves if linked to hr.holiday with holiday_id, then the resource.calendar.leaves should have the same time_type as its holiday.
See https://github.com/odoo/odoo/blob/12.0/addons/hr_holidays/models/hr_leave.py#L544

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 4 above)
Comment (MiquelRForgeFlow) [reply addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:16]: And thus I should do something?

Comment (pedrobaeza) [reply addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:16]: Not yet. Only for taking into account for the future.

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 3 above)
Comment (pedrobaeza) [reply addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:8]: OK, then we reassign in that module.

File: addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt (hunk 2 above)
Comment (pedrobaeza) [reply addons/resource/migrations/12.0.1.1/openupgrade_analysis_work.txt:5]: No need for that in the proper field. It's a float field.

**Comment by pedrobaeza** [comment]
I have finished the scripts with the remaining points.

@mreficent several things for remembering when doing other migration scripts:

* no need since v10 to put `use_env=True`. That's the default.
* Use `openupgrade.logged_query` for most of the SQL for having the proper trace.
* You can use the first line for the SQL expression as it will be output correctly on log without the extra line break.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**